packages/edit-widgets/src/blocks/widget-area/edit/index.js

```diff
@@ -20,36 +21,47 @@ export default function WidgetAreaEdit( {
 	className,
 	attributes: { id, name },
 } ) {
-	const isOpen = useSelect(
-		( select ) =>
-			select( 'core/edit-widgets' ).getIsWidgetAreaOpen( clientId ),
+	const { isOpen, isDraggingBlocks } = useSelect(
+		( select ) => ( {
+			isOpen: select( 'core/edit-widgets' ).getIsWidgetAreaOpen(
+				clientId
+			),
+			isDraggingBlocks: select( 'core/block-editor' ).isDraggingBlocks(),
+		} ),
 		[ clientId ]
 	);
 	const { setIsWidgetAreaOpen } = useDispatch( 'core/edit-widgets' );
+	const openOnDragOver = useCallback( () => {
+		if ( ! isOpen && isDraggingBlocks ) {
```

Comment:
I don't think we should worry about `isDraggingBlocks`, it should also open for dragged files and HTML so that users can upload/add things like images that way.